lib/src/edhoc.rs

```diff
@@ -24,22 +26,15 @@ pub fn edhoc_exporter(
         _th_3,
     ) = state;
 
-    let mut output: BytesMaxBuffer = [0x00; MAX_BUFFER_LEN];
-    let mut error = EDHOCError::UnknownError;
-
-    if current_state == EDHOCState::Completed {
-        output = edhoc_kdf(&prk_exporter, label, context, context_len, length);
-        Ok((state, output))
-    } else {
-        Err(EDHOCError::WrongState)
-    }
+    edhoc_kdf(crypto, &prk_exporter, label, context, context_len, length)
```

Comment:
It's great how concise it became. Do we still need to decompose `state`?

Reply:
As long as we have the tuple structs it's more readable that way -- otherwise we'd have `edhoc_kdf(crypto, &state.7, label, context, context_len, length)`.

So I think the fix here is to refactor the tuple structs, which a) is best done in a flag day operation, and b) may be easiest with some kind of refactoring tooling (one of the few times an IDE can shine).